src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -462,6 +463,15 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 					       devc->rle_count);
 					devc->raw_sample_size = 0;
 
+					if (devc->trigger_at_smpl !=
+						    OLS_NO_TRIGGER &&
+					    devc->trigger_rle_at_smpl_from_end ==
+						    OLS_NO_TRIGGER &&
+					    (unsigned int)devc->trigger_at_smpl ==
+						    devc->cnt_rx_raw_samples)
+						devc->trigger_rle_at_smpl_from_end =
+							devc->cnt_samples;
+
 					/*
```

Comment:
This if-condition is really hard to read. And unless I missed something, it is needed a second time later. I'd suggest an inline function here like "is_<some_good_description>_trigger(devc)"

Reply:
Good point! I extracted the ugliness to `set_rle_trigger_point_if_unset`